File: Segundo/PDOO/CivitasJava/src/main/java/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author jmramirez
 */
public class CasillaImpuesto extends Casilla {
    
    private float importe;
    
    CasillaImpuesto(float cantidad, String n) {
        super(n);
        importe = cantidad;
    }
    
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
    public String toString() {
        String ret = "la casilla ";
        ret += this.nombre;
        ret += ", de tipo Impuesto.";
        return ret;
    }
    
}

File: Segundo/PDOO/CivitasRuby/Civitas/CasillaImpuesto.rb
# encoding:utf-8

require_relative "Casilla"

module Civitas

    class CasillaImpuesto < Casilla

        attr_reader :cantidad

        def initialize(n, cant)
            super(n)
            @cantidad=cant
        end

        def informe(actual, todos)
            Diario.instance.ocurre_evento("El jugador " + todos.at(actual).nombre + " ha caído en una casilla de tipo Impuesto.")
        end

        def recibe_jugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                todos.at(actual).pagaImpuesto(@cantidad)
            end
        end

        def toString
            return "La casilla actual es " + @nombre + " y es de tipo Impuesto.\n"
        end

    end

end